python/kiss_icp/datasets/mcap.py

```diff
@@ -104,29 +61,37 @@ def __getitem__(self, idx):
     def __len__(self):
         return self.n_scans
 
+    def _get_n_scans(self) -> int:
+        return [
+            count
+            for (id, count) in self.summary.statistics.channel_message_counts.items()
+            if self.summary.channels[id].topic == self.topic
+        ][0]
```

Comment:
Might want to use `sum(...)` instead of `[...][0]`?

Reply:
This is a tricky one... Is possible to have multiple channels with the same topic? the `sum` keyword give me the wrong impression, that we are accumulating the number of messages.

The current solution looks horrible anyway, this I agree :rofl:

Reply:
Yes, it's allowed, maybe more likely with conversion from ROS 1 bags where each connection is converted to a channel. I'm not sure off the top of my head if ROS 2 recordings consolidate topics.

Reply:
Then the sum thing make sense, would you mind opening a PR, so I can give "some credit"? Otherwise, I can also do it, just let me know

Reply:
https://github.com/PRBonn/kiss-icp/pull/114

Reply:
Thanks!